runtime/src/main/cpp/Cleaner.cpp

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2010-2020 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license
+ * that can be found in the LICENSE file.
+ */
+
+#include "Cleaner.h"
+
+#include "Memory.h"
+#include "Runtime.h"
+
+// Defined in Cleaner.kt
+extern "C" void Kotlin_CleanerImpl_clean(KRef thiz);
+extern "C" void Kotlin_CleanerImpl_shutdownCleanerWorker(bool);
+
+namespace {
+
+bool cleanersDisabled = false;
+
+void disposeCleaner(KRef thiz) {
+    if (atomicGet(&cleanersDisabled)) {
+        if (Kotlin_cleanersLeakCheckerEnabled()) {
+            konan::consoleErrorf(
+                    "Cleaner %p was disposed after main()\n"
```

Comment:
It's not always `main()`. Maybe use more general wording like `during program exit`?